shared/src/main/scala/amf/core/model/domain/DataModel.scala

```diff
@@ -13,18 +13,16 @@ import amf.core.resolution.VariableReplacer
 import amf.core.utils._
 import amf.core.vocabulary.Namespace.Data
 import amf.core.vocabulary.{Namespace, ValueType}
-import org.yaml.model.{YPart, YSequence}
+import org.yaml.model.{YPart, YScalar, YSequence}
 
 import scala.collection.mutable
 
 /**
   * Base class for all dynamic DataNodes
   */
-abstract class DataNode(annotations: Annotations) extends DomainElement {
+abstract class DataNode(annotations: Annotations) extends DomainElement with NamedDomainElement {
 
-  def name: StrField = fields.field(Name)
-
-  def withName(name: String): this.type = set(Name, name)
+  override protected def nameField: Field = DataNodeModel.Name // ??
```

Comment:
Is this correct? Asking because of the "??" comment

Reply:
I'm not sure if is correct, If you ask me I would say no.
Watch is the sense of a name at a scalar node? I think only objects could be named, but as I wasn't sure added the comment.

Reply:
lgtm